Dockerfile-cni-plugin

```diff
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BUILDPLATFORM=linux/amd64
+
+# Precompile key slow-to-build dependencies
+FROM --platform=$BUILDPLATFORM golang:1.18-alpine as golang
+WORKDIR /linkerd-build
+COPY --link go.mod go.sum .
```

Comment:
TIL of `--link`. If I understand it, `--link` basically produces its own layer which at the end is linked together with the rest of the dockerfile. Without it, we'd depend on the previous layer.

This means that caching should work better since we won't need to compute any diffs, or we won't need to compute the previous layers? Not sure my question makes sense

Reply:
That's my understanding as well. I'm following what @olix0r has done in the `proxy-init` Dockerfile so I'm going to invite him to chime in here.

Reply:
I can research it a little bit and post my findings somewhere 👍🏻